packages/loopring_v3/contracts/impl/relayer/FastWithdrawal.sol

```diff
@@ -0,0 +1,92 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "./RelayerData.sol";
+
+import "../../iface/IExchangeV3.sol";
+import "../../iface/ICookieJarContract.sol";
+
+import "../../lib/MathUint.sol";
+import "../../thirdparty/ECDSA.sol";
+
+/// @title An Implementation of IRelayer.
+/// @author Brecht Devos - <brecht@loopring.org>
+contract FastWithdrawal is RelayerData
```

Comment:
FastWithdrawal should be Relayer module right?  so it doesn't need to extend RelayerData.

Reply:
This was to allow sharing data between modules. And because the modules are called with delegatecall (to keep `msg.sender` the Relayer contract) they will share state (it's essentially a proxy with many implementation contracts). So it depends on what the process will be for adding a module, but it's certainly not needed. There are many ways to do it.